Automattic/wp-calypso
# Earn: Convert controller to TypeScript

## Proposed Changes

Converts the Earn controller to TypeScript. Part of a broader effort to update the Earn code.

## Testing Instructions



Checkout this branch, go to `http://calypso.localhost:3000/earn/YOURDOMAIN`, click around different tabs and sections and confirm everything works as before. As this refactor is for the controller, the main thing we're checking is that navigation between tabs and sections works without issue.
